editors/code/src/test_explorer/run_or_debug.ts

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable no-console */
+import * as vscode from "vscode";
+import * as toolchain from "../toolchain";
+import { testController } from ".";
+import { spawn } from "child_process";
+import { assert } from "../util";
+import { createArgs, prepareEnv } from "../run";
+import { getRunnableByTestItem } from "./discover_and_update";
+import { TestControllerHelper } from "./TestControllerHelper";
+import { startDebugSession } from "../debug";
+import { raContext } from "../main";
+import { RustcOutputAnalyzer } from "./RustcOutputAnalyzer";
+
+export async function runHandler(
+    request: vscode.TestRunRequest,
+    token: vscode.CancellationToken
+) {
+    // TODO: Never run tests concurrently in client side.
```

Comment:
it's not entirely clear to me as to why this limitation exists. if a user wants to run multiple tests, can the client not simply pass multiple arguments to the test runner?

Reply:
This says, not allow user to click two "Run" button quickly, or , when a test is running, not allow another test is runed.

I am not sure what would happen if we allow this, because many process might download same thing meanwhile, or is there any lock file for test command.

And, in fact, we do not allow people run multi tests(but they could run tests for a package or for a whole test module) anyway for now. When user choose multi test-items and click run, we will log some message like "You could run one test item at one time."